plugins/modules/pnp_intent.py

```diff
@@ -609,86 +609,61 @@
     }
 """
 
-import copy
 from ansible.module_utils.basic import AnsibleModule
 from ansible_collections.cisco.dnac.plugins.module_utils.dnac import (
-    DNACSDK,
+    DnacBase,
     validate_list_of_dicts,
-    log,
-    get_dict_result,
+    get_dict_result
 )
 
 
-class DnacPnp:
+class DnacPnp(DnacBase):
     def __init__(self, module):
-        self.module = module
-        self.params = module.params
-        self.config = copy.deepcopy(module.params.get("config"))
-        self.have = []
-        self.want = []
-        self.diff = []
-        self.validated = []
-        dnac_params = self.get_dnac_params(self.params)
-        log(str(dnac_params))
-        self.dnac = DNACSDK(params=dnac_params)
-        self.log = dnac_params.get("dnac_log")
-
-        self.result = dict(changed=False, diff=[], response=[], warnings=[])
-
-    def get_state(self):
-        return self.params.get("state")
+        super().__init__(module)
 
     def validate_input(self):
-        pnp_spec = dict(
-            template_name=dict(required=True, type='str'),
-            project_name=dict(required=False, type='str', default="Onboarding Configuration"),
-            site_name=dict(required=True, type='str'),
-            image_name=dict(required=True, type='str'),
-            golden_image=dict(required=False, type='bool'),
-            deviceInfo=dict(required=True, type='dict'),
-            pnp_type=dict(required=False, type=str, default="Default")
+        """Validate the fields provided in the playbook"""
+
+        if not self.config:
+            self.msg = "config not available in playbook for validation"
+            self.status = "success"
+            return self
+
+        pnp_spec = {
+            'template_name': {'required': True, 'type': 'str'},
```

Comment:
'template_name': {'type': 'str', 'required': True},

Reply:
This is done